modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -1120,3 +1127,110 @@ class EndpointListController extends AbstractFormController
 				@serviceReturn = null
 			dataType: 'json'
 
+	getCurrentEndpoints: => 
+		# get the current endpoints and their values for the protocol from @collection
+
+		# create holders for each one we want to collection 
+		endpointNames = []
+		endpointUnits = []
+		endpointDataTypes = []
+		endpointConc = []
+		endpointConcUnits = []
+		endpointTime = []
+		endpointTimeUnits = []
+		endpointHidden = []
+
+		for lsState in @collection 
```

Comment:
This would all greatly benefit from using Backbone collection methods and State class methods:

https://github.com/mcneilco/acas/blob/7eba2bfe3ff6ec0aa039ab775cc0b045857b6ef1/modules/ServerAPI/src/client/Label.coffee#L201